src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.Api/Controllers/ProcessController.cs

```diff
@@ -243,6 +221,26 @@ private async Task<Instance> UpdateProcessAndDispatchEvents(Instance instance, P
             }
         }
 
+        private async Task<bool> CanTaskBeEnded(Instance instance, string currentElementId)
```

Comment:
From the naming 'CanTaskBeEnded' this looks like a query method, but reading the code it looks like it update things as well ie. it has side effects. Would consider then to rename the method to 'TryEndTask'

Reply:
The ending of the task happens after this check. Keeping name.

Reply:
Yes, but I was think more on the call to the ValidateAndUpdateProcess method - and specifically the Update part. Does this method call actually change values in the process?

Reply:
It updates validation status flag. But not the process itself. Makes it possible to change process task.